### PR TITLE
configcore: issue a warning on core16 when journal.persistent option is set

### PR DESCRIPTION
**Pull request by stolowski**
Issue a warning if persistent journal is set on core16 where journald cannot be reloaded.
Followup to #8602

### CODE REVIEW

File: overlord/configstate/configcore/journal.go [hunk 1]
```diff
@@ -100,10 +100,29 @@ func handleJournalConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error
...
+			if err := sysd.Kill("systemd-journald", "USR1", ""); err != nil {
+				return err
+			}
+		} else {
+			cfg, ok := tr.(config.Conf)
+			if ok {
+				st := cfg.State()
+				st.Lock()
+				defer st.Unlock()
+				st.Warnf("manual restart required for journal.persistent change to take effect")
```
Comment (zyga) [inline overlord/configstate/configcore/journal.go:123]: Reading this it is unclear what needs to be restarted, how about this:
```suggestion
				st.Warnf("system restart required for journal.persistent change to take effect")
```

File: systemd/systemd.go [hunk 2]
```diff
@@ -129,6 +131,40 @@ func Available() error {
...
+// Version returns systemd version
+func Version() (int, error) {
+	out, err := systemctlCmd("--version")
+	if err != nil {
+		return 0, err
+	}
+
+	// systemd version outpus is two lines - actual version and a list
+	// of features, e.g:
+	// systemd 229
```
Comment (mvo5) [inline systemd/systemd.go:143]: I think this example should also include the form: `systemd 229 (229.1-4ubuntu1)` to show why this is done in this relatively complicated way

Comment (stolowski) [reply systemd/systemd.go:143]: Yes, this is part of prerequsite PR which I forgot to merge here, pushed now.

File: tests/main/core-persistent-journal/task.yaml [hunk 3]
```diff
@@ -32,7 +32,14 @@ execute: |
 
     # on core16 journald is not reloaded, so changes are not visible until
     # rebooted.
+<<<<<<< HEAD
```
Comment (mvo5) [inline tests/main/core-persistent-journal/task.yaml:35]: There is a conflict here

Comment (stolowski) [reply tests/main/core-persistent-journal/task.yaml]: Thanks, silly me, fixed now.

**Comment by stolowski** [comment]
<pedronis> pstolowski: it's blocked becausse mvo made the good point that we should auto-remove the warning if there's a reboot
<pedronis> pstolowski: that needs a bit of infra work I fear

**Comment by mvo5** [comment]
Closing for now as this needs further work. Let's get back to it at some later point.

**Pull request closed without merging**